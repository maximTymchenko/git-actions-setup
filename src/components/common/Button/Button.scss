@import '../../../styles/vars';

.button {
    // width: 100%;
    width: 210px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    cursor: pointer;
    font-family: 'OktaNeue-Bold';
    font-size: 15px;
    font-weight: bold;
    letter-spacing: 0.5px;
    @include transition__shadow;
    font-size: 15px;
    margin-right: 20px;
    box-shadow: 3px 3px 1px 1px rgba(57, 156, 203, 0.4);
    padding: 8px;
    outline: none;
    border: 2px solid $secondary;
    color: $secondary;

    @media screen and (max-width: $mobile) {
        font-size: 12px;
    }

    &--mt {
        margin-top: 15px;
    }

    &--large {
        width: 220px;
        margin: 0 auto;

        @media screen and (max-width: $tablet) {
            margin-bottom: 20px;
        }
    }

    &--clear {
        width: 100px;
        background-color: transparent;
        border: none;
        box-shadow: none;
        font-family: 'OktaNeue-SemiBold';
        font-size: 15px;
        font-weight: 600;
        letter-spacing: 0.3px;
        color: $text__body;
    }

    &--apply {
        width: 80px;
        background-color: transparent;
        border: none;
        box-shadow: none;
        font-family: 'OktaNeue-SemiBold';
        font-size: 15px;
        font-weight: 600;
        letter-spacing: 0.3px;
        color: $secondary;

        &--reverse {
            flex-direction: row-reverse;
            width: 80px;
            background-color: transparent;
            border: none;
            box-shadow: none;
            font-family: 'OktaNeue-SemiBold';
            font-size: 15px;
            font-weight: 600;
            letter-spacing: 0.3px;
            color: $secondary;
        }
    }

    &--mobile {
        @media screen and (max-width: $large__mobile) {
            width: 125px;
            // height: 50px;
        }
    }

    &--watch {
        justify-content: flex-start;
        box-shadow: none;
        display: flex;
        align-items: center;
        background: none;
        border: none;
        padding-right: 12px;

        & img {
            margin-left: 10px;
        }

        @media screen and (max-width: $mobile) {
            display: none;
        }
    }

    &--circle {
        width: 155px;
        height: 155px;
        box-shadow: none;
        display: flex;
        align-items: center;
        background: $background__secondary;
        border-radius: 50%;
        border: none;
        padding-left: 25px;
        @include transition;

        @media screen and (max-width: $mobile) {
            margin-left: 100px;
            position: relative;
            left: -20px;
        }

        & img {
            margin-left: 10px;
        }

        &:hover {
            background-color: $primary__light--70;
        }
    }

    &--small-circle {
        width: 110px;
        height: 110px;
        margin-left: 80px;
        justify-content: flex-start;
        box-shadow: none;
        display: flex;
        align-items: center;
        background: $background__secondary;
        border-radius: 50%;
        border: none;
        padding-left: 20px;
        @include transition;
        // @include transition__shadow;

        & img {
            margin-left: 10px;
        }

        &:hover {
            background-color: $primary__light--70;
        }
    }

    &--secondary {
        background-color: $primary__light;

        &:hover {
            background-color: $brand__primary;
        }

        &:active {
            @include button__press;
            box-shadow: 3px 3px 1px 1px rgba(57, 156, 203, 0);
            background-color: $primary__dark;
        }
    }

    &--primary {
        background-color: $background__primary;

        &:hover {
            background-color: $background__secondary;
        }

        &:active {
            @include button__press;
            box-shadow: 3px 3px 1px 1px rgba(57, 156, 203, 0);
            background-color: $text__body--light;
        }
    }

    &--reverse {
        flex-direction: row-reverse;

        background-color: $background__primary;

        &:hover {
            background-color: $background__secondary;
        }

        &:active {
            @include button__press;
            box-shadow: 3px 3px 1px 1px rgba(57, 156, 203, 0);
            background-color: $text__body--light;
        }
    }
}
