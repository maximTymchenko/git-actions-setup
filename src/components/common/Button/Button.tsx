import React, { FC } from 'react';
import './Button.scss';
import ButtonIcon from '../ButtonIcon';

interface IButtonProps {
    children: string;
    icon: any;
    type: string;
    desc?: string;
    mobile?: string;
    function?: () => void;
    subtype?: string;
    iconType?: string;
}

const Button: FC = (props: IButtonProps) => {
    return (
        <button
            data-testid="button"
            className={`button button--${props.type} button--${props.mobile} button--${props.subtype}`}
        >
            {props.children}
            <ButtonIcon type={props.iconType} icon={props.icon} desc={props.desc} />
        </button>
    );
};

export default Button;
