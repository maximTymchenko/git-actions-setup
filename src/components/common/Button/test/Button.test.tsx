import React from 'react';
import Button from '../Button';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<Button></Button>);
    expect(getByTestId('button')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<Button children="Test"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});
