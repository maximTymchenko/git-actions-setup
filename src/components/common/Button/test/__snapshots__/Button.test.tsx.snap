// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<button
  className="button button--undefined button--undefined button--undefined"
  data-testid="button"
>
  Test
  <img
    className="button__icon"
    data-testid="button__icon"
  />
</button>
`;
